src/main/java/ch/njol/skript/expressions/ExprValueWithin.java

```diff
@@ -0,0 +1,101 @@
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.classes.ClassInfo;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.WrapperExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.util.Kleenean;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Value Within")
+@Description("Gets the value within objects. Usually used with variables to get the value they store rather than the variable itself, " +
+	"or with lists to get the values of a type.")
+@Examples({
+	"set {_entity} to a random entity out of all entities",
+	"delete entity within {_entity} # This deletes the entity itself and not the value stored in the variable",
+	"",
+	"set {_list::*} to \"something\", 10, \"test\" and a zombie",
+	"broadcast the strings within {_list::*} # \"something\", \"test\""
+})
+@Since("INSERT VERSION")
+public class ExprValueWithin extends WrapperExpression<Object> {
+
+	static {
+		Skript.registerExpression(ExprValueWithin.class, Object.class, ExpressionType.SIMPLE, "[the] (%-*classinfo%|value[s]) (within|in) %~objects%");
+	}
+
+	@Nullable
+	private ClassInfo<?> classInfo;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		classInfo = exprs[0] == null ? null : ((Literal<ClassInfo<?>>) exprs[0]).getSingle();
+		Expression<?> expr = exprs[0] == null ? exprs[1] : exprs[1].getConvertedExpression(classInfo.getC());
+		if (expr == null)
+			return false;
+		setExpr(expr);
+		return true;
+	}
+
+	@Nullable
+	private ClassInfo<?> returnTypeInfo;
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (returnTypeInfo == null)
+			returnTypeInfo = Classes.getSuperClassInfo(getReturnType());
+		Changer<?> changer = returnTypeInfo.getChanger();
```

Comment:
Can we just cache this changer instead?

Reply:
Well the value within the variable could change at any time and it could use a different changer :shrug:

Reply:
You can in the init method. The `Classes.getSuperClassInfo(getReturnType());` will be valid after setting the `setExpr(expr);`

Reply:
Wouldn't that be a bit redundant? Because then we'd be caching the changer if every time the expression is used when we only need to cache it if the EffChange effect is used

Reply:
acceptChange and init method are only ran during parse time, not constantly at runtime. Sure this is fine, you only use the changer in the change method, but we typically setup the fields in the init method rather than the acceptChange.

You also don't need a null check here as it's certain that it'll be set on parse.